src/main/java/net/smartcosmos/dao/relationships/RelationshipDao.java

```diff
@@ -28,6 +33,19 @@
     Optional<RelationshipResponse> findByUrn(String accountUrn, String urn);
 
     /**
+     * Finds a relationship matching specified entity, related entity and type.
+     *
+     * @param accountUrn
+     * @param relationshipLookupSpecific
+     * @return
+     * @throws ConstraintViolationException if the {@link RelationshipLookupSpecific} violates constraints enforced by the persistence service
+     */
+    Optional<RelationshipResponse> findSpecific(
+            String accountUrn,
+            RelationshipLookupSpecific relationshipLookupSpecific)
+        throws ConstraintViolationException;
```

Comment:
Lookup won't throw a `ConstraintViolationException`. Only `create()` and `update()` do that. Lookup would just don't find anything.